tutorial/tests/testsuite.py

```diff
@@ -340,10 +242,30 @@ def ipytest(self, line: str, cell: str):
             )
 
 
+def retrieve_functions(
+    all_functions: Dict, node: object, called_functions: Set[str]
+) -> Set[object]:
+    """Recursively walk the AST tree to retrieve all function definitions in a file"""
+
+    for n in ast.walk(node):
+        if isinstance(n, ast.Call) and hasattr(n.func, "id"):
```

Comment:
You can use a match statement here, for example


```suggestion
match node:
    case ast.Call(ast.Name(id=name), args, keywords):
        """do your work with name here"""
```

```